topiary-core/src/atom_collection.rs

```diff
@@ -883,13 +896,62 @@ impl AtomCollection {
         }
     }
 
+    /// Separate post processing of capitalisation, to avoid confusion around whitespacing.
+    fn post_process_capitalization(&mut self) {
+        let mut case_context: Vec<Capitalisation> = Vec::new();
+        for atom in &mut self.atoms {
+            match atom {
+                Atom::CaseBegin(case) => match case {
+                    Capitalisation::UpperCase => {
+                        case_context.push(Capitalisation::UpperCase);
+                        *atom = Atom::Empty;
+                    }
+                    Capitalisation::LowerCase => {
+                        case_context.push(Capitalisation::LowerCase);
+                        *atom = Atom::Empty;
+                    }
+                    Capitalisation::Pass => {
+                        case_context.push(Capitalisation::Pass);
+                        *atom = Atom::Empty;
+                    }
+                },
```

Comment:
Would this be simpler, or is there a need to make each branch explicit?

```suggestion
                Atom::CaseBegin(case) => {
                    case_context.push(case.clone());
                    *atom = Atom::Empty;
                }
```

I don't think it matters too much that we're cloning `case`.